### ISSUE TITLE
Doubtful locations in distribution plots

### ISSUE THREAD

**Issue by alrichardbollans** [matildabrown/rWCVP#46]
Plotting the following distributions with:
```
library(rWCVP)

distribution <- wcvp_distribution("Gentianales", taxon_rank="order", location_doubtful = FALSE)
wcvp_distribution_map(distribution)
distribution <- wcvp_distribution("Rubiaceae", taxon_rank="family", location_doubtful = FALSE)
wcvp_distribution_map(distribution)
distribution <- wcvp_distribution("Aidia pycnantha", taxon_rank="species", location_doubtful = FALSE)
wcvp_distribution_map(distribution)
```

returns:
Gentianales:
![Gents_plot](https://user-images.githubusercontent.com/38588335/228800589-1e97930a-972d-4624-af06-3f51be8db466.png)

Rubiaceae:
![Rubiaceae](https://user-images.githubusercontent.com/38588335/228800661-5f7f9412-fbe9-407c-8124-11edcb71e7e8.png)

Aidia pycnantha:
![Aidia pycnantha](https://user-images.githubusercontent.com/38588335/228800701-49d345e3-e809-4332-8c85-43c6f4d5b233.png)

The maps for the family and order this produces seem incorrect to me i.e. there are certainly gentianales (not doubtful!) in all/most of africa and south america, indonesia, china etc.. For example, Aidia pycnantha in Rubiaceae is found in east China and this shows up in the Rubiaceae and Aidia plots but not the Gentianales plot. Similarly, more regions are marked in the Rubiaceae plot than the Gentianales plots, which seems counterintuitive and there are also lots of regions in the Rubiaceae plot which I think should be marked as 'native'.

Maybe this is the intended behaviour, but I suspect `location_doubtful = FALSE` is removing those regions that are doubtful for *any* species when it should(?) remove those regions that are doubtful for all species.

**Comment by barnabywalker** [comment]
I think we discussed this before, then maybe never implemented a solution.

The case of deciding between native and the other occurrence types seems obvious (so e.g. if some species in a genus/family/order are native to a region but some are location_doubtful/introduced/extinct it should show as native), but not sure how to decide between the others.

I feel like the priority should run `native > introduced > extinct > location_doubtful` - does that sound right?

**Comment by alrichardbollans** [comment]
I think this really depends on individual purpose and above the species level it gets complicated having a mix of statuses. In general I would agree that `native > introduced`. However, as far as I understand, where native and introduced are separate states, `extinct` and `location_doubtful` are modifiers of those states e.g. a species can be doubtful in a location where is is presumed to be introduced, or extinct in a location where is was native/introduced.

I would be inclined to just label regions based on native/introduced states and maybe a cross-hatching colour for where both native and introduced species are. I think it gets quite messy trying to include `extinct` and `doubtful` info above the species level.

**Comment by barnabywalker** [comment]
I think our intention with `wcvp_distribution` was that it would replicate the distributions displayed on POWO.

Looking at POWO, for individual species, it looks like:
- where a species is both `extinct` and `location_doubtful` in a region, it is displayed as `location_doubtful` (e.g. [*Silene polypetala*](https://powo.science.kew.org/taxon/urn:lsid:ipni.org:names:236357-2))
- where a species is both `extinct` and `introduced` in a region, it is displayed as `introduced` (e.g. [*Bulboschoenus glaucus*](https://powo.science.kew.org/taxon/urn:lsid:ipni.org:names:982498-1))
- where a species is both `introduced` and `location_doubtful`, it is displayed as `introduced` (e.g. [*Euphorbia multinodis*](https://powo.science.kew.org/taxon/urn:lsid:ipni.org:names:101760-2))

For genera:
- if there is at least one `native` species in a region, the genus is `native` (e.g. [*Acalypha*](https://powo.science.kew.org/taxon/urn:lsid:ipni.org:names:327639-2) is native to Alabama, Florida and Louisiana but [*Acalypha alopecuroidea*](https://powo.science.kew.org/taxon/urn:lsid:ipni.org:names:337150-1) is introduced in all 3 regions)
- a region is only shown as `extinct` if all species that occur there are extinct (e.g. [*Aldrovanda*](https://powo.science.kew.org/taxon/urn:lsid:ipni.org:names:100027-3))
- a region is shown as `location_doubtful` if all species that occur there are listed as doubtful (e.g. [*Aldrovanda*](https://powo.science.kew.org/taxon/urn:lsid:ipni.org:names:100027-3))

So, I'll fix `wcvp_distribution` to follow that for the taxonomy from family upwards, as well, I think?

**Comment by matildabrown** [comment]
That sounds perfect @barnabywalker  - I love the idea of cross-hatching @alrichardbollans  but the intention is definitely to replicate the POWO mapping (though other themes in future versions is a tempting idea...). 

Could you add a test for your fix too, Baz?

**Comment by alrichardbollans** [comment]
@barnabywalker yep this sounds good!

**Comment by barnabywalker** [comment]
Just adding tests for this, but I think it should be fixed now. Maps below - Gentianales and Rubiaceae showing up pretty much everywhere, which I think is right?

Gentianales:
![gentianales-distribution](https://user-images.githubusercontent.com/6917951/228893516-1b3463e9-3859-4903-9499-8c0068b00351.png)

Rubiaceae:
![rubiaceae-distribution](https://user-images.githubusercontent.com/6917951/228893586-60233f93-06ce-4187-9649-64835908e147.png)

_Aidia pycnantha_:
![aidia-pycnantha-distribution](https://user-images.githubusercontent.com/6917951/228893735-226c8f11-a628-42f2-b6bf-3ce23275ba25.png)

**Comment by barnabywalker** [comment]
I've added the fix and an associated test in #48, but it would be good for someone to double-check it works as expected before merging.
